File: 1-moderate/prime_numbers/prime_numbers.js
var fs  = require("fs");

fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line != "") {
        var number = parseInt(line);
        var primes = [];
        for(var i = 0; i <= number; i++) {
            if (isPrime(i)) {
                primes.push(i);
            }
        }
        console.log(primes.join(","));
    }
});

function isPrime(number) {
    if (number <= 1) {
        return false;
    }
    var limit = Math.floor(Math.sqrt(number));
    for(var i = 2; i <= limit; i++) {
        if (number % i === 0) {
            return false;
        }   
    }
    return true;
}

File: 1-moderate/prime_numbers/prime_numbers.rb
class Prime

  def self.prime?(number)
    return false if number <= 1 || !number.integer?
    for i in 2..Math.sqrt(number).to_i
      return false if (number % i == 0)
    end
    true
  end

  def self.each(size)
    primes = self.sequence
    return primes unless block_given?
    size.times { yield primes.next }
  end

  def self.first(size)
    self.sequence.take(size)
  end

  def self.up_to(number)
    primes = self.sequence
    result = []
    loop do
      prime = primes.next
      if prime <= number
        result << prime
      else
        break
      end
    end
    result
  end

private
  def self.sequence
    Enumerator.new do |y|
      n = 2
      loop do
        y << n if self.prime?(n)
        n += 1
      end
    end
  end

end

ARGF.each do |line|
  line = line.strip
  next if line.empty?
  puts Prime.up_to(line.to_i).join(",")
end